File: Python/JCT/greedy/maxCard2.py
N, M = map(int, input().split(' '))

def min(a, b):
    return a if a < b else b

def max(a, b):
    return a if a > b else b

max_val = 1
for row in range(0, N):
    min_val = 10001
    data = list(map(int, input().split()))
    for num in data:
        min_val = min(min_val, num)
    max_val = max(min_val, max_val)

print(max_val)

File: Java/JCT/greedy/MaxCard2.java
package greedy;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class MaxCard2 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());

        int maxVal = 1;
        for (int i = 0; i < N; i++) {
            int minVal = 10001;
            st = new StringTokenizer(br.readLine());
            for (int j = 0; j < M; j++) {
                int nextInt = Integer.parseInt(st.nextToken());
                minVal = min(minVal, nextInt);
            }
            maxVal = max(maxVal, minVal);
        }

        System.out.println(maxVal);
        br.close();
    }

    private static int min(int a, int b) {
        return a < b ? a : b;
    }

    private static int max(int a, int b) {
        return a > b ? a : b;
    }
}
